### DISCUSSION TITLE
umi 的 entry 不允许改的情况下，怎么在加载启动 umi 之前替换 require 实现？

### DISCUSSION THREAD

**Discussion by biaomingzhong** [umijs/umi#6652] [Q&A] [answered]
当使用 umi 开发 electron 应用，如果要使用类似于 atom 的 v8 snapshot 的特性，需要运行时替换 require 的实现，去使用  snapshotResult 这个变量中的模块。

类似于 atom 里面的实现：

https://github.com/atom/atom/blob/0dd75df54077414fdd835d802c86dd3603ac7cca/static/index.js#L60

**Comment by biaomingzhong** [comment]
看到之前有些 issue 里是有提到不能修改 entry 的，https://github.com/umijs/umi/issues/4428#issuecomment-694772496

**Comment by biaomingzhong** [comment]
其实就是运行时替换 require 的实现，这个逻辑很重要，要使用 snapshot 就是这样实现的

**Comment by biaomingzhong** [comment, marked as answer]
解决了，通过自己创建一个 umi plugin，再配一下 config.ts 里的`plugins` 引入，使用 `api.addEntryCodeAhead` 引入需要的代码在最前面

**Reply by xiaohuoni** [reply]
直接在 global.ts 中引入，应该也行
